### ISSUE TITLE
Generated reverse route method should have empty parentheses () depending on routes file

### ISSUE THREAD

**Issue by gakuzzzz** [playframework/playframework#10363]
### Play Version

2.8.2

### API

Scala

### Operating System

Not related to this problem

### JDK

Not related to this problem

### Library Dependencies

non

### Expected Behavior

No warning in Scala2.13.3.

### Actual Behavior

1. controller

```scala
package controllers

import javax.inject._
import play.api._
import play.api.mvc._

@Singleton
class HomeController @Inject()(val controllerComponents: ControllerComponents) extends BaseController {

  def index = Action { implicit request: Request[AnyContent] =>
    Ok("")
  }

  def redirect = Action { implicit request: Request[AnyContent] =>
    Redirect(routes.HomeController.index)
  }

}
```

2. compile

Warning is displayed.

```
[warn] D:\workspace\example\app\controllers\HomeController.scala:15:36: Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method index,
[warn] or remove the empty argument list from its definition (Java-defined methods are exempt).
[warn] In Scala 3, an unapplied method like this will be eta-expanded into a function.
[warn]     Redirect(routes.HomeController.index)
[warn]                                    ^
[warn] one warning found
```

### Related resources

- https://github.com/scala/scala/pull/8833
- https://github.com/scala/scala/releases/tag/v2.13.3

**Comment by mkurz** [comment]
I think you just need to change
```scala
Redirect(routes.HomeController.index)
```
to
```scala
Redirect(routes.HomeController.index())
```
However I think we also need to fix Play's routes generator as well: #10364

**Comment by gakuzzzz** [comment]
Hi, @mkurz 

Thanks, I think so too.

When the `index` was defined as `def index() = Action{...}`, we need to change to `Redirect(routes.HomeController.index())`.
However the `index` method has no side-effect. so it is defined as `def index = Action{...}`.
We expect that Play's routes generator generates no-parentheses method.

regards,

**Comment by mkurz** [comment]
Fix is here #10671
